main_task/src/custom_implementations/ArrayListOfStudents.java

```diff
@@ -0,0 +1,108 @@
+package custom_implementations;
+import main.Student;
+
+public class ArrayListOfStudents {
+    private int initialCapacity;
+    private int indexOfTop = -1;
+    private Student[] students;
+
+    public ArrayListOfStudents(int initialCapacityOfArray) {
+        initialCapacity = initialCapacityOfArray;
+        students = new Student[initialCapacity];
+    }
+
+    public ArrayListOfStudents(ArrayListOfStudents anotherArrayListOfStudents) {
```

Comment:
Need to check whether initial capacity is negative and throw an IllegalArgumentException